jit.js

```diff
@@ -692,7 +932,12 @@ to single-step.
     generateLabel: function() {
         // remember label position for deleteUnneededLabels()
         this.sourceLabels[this.prevPC] = this.source.length;
-        this.source.push("case ", this.prevPC, ":\n");
+        if (!this.suppressNextLabel) {
+            this.source.push("case ", this.prevPC, ":\n");
+        } else {
+            this.source.push("\n");
+        }
+        this.suppressNextLabel = false;
```

Comment:
The main reason was, that we did not understand the label-removing mechanism and it was giving us a lot of headaches. A quick explanation of how it works would be awesome!